src/Storage/Query/Handler/SelectQueryHandler.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bolt\Storage\Query\Handler;
+
+use Bolt\Entity\Content;
+use Bolt\Storage\Query\ContentQueryParser;
+use Bolt\Storage\Query\QueryResultset;
+use Bolt\Storage\Query\SelectQuery;
+
+/**
+ *  Handler class to perform select query and return a resultset.
+ */
+class SelectQueryHandler
+{
+    /**
+     * @return Content[]|Content|false
+     */
+    public function __invoke(ContentQueryParser $contentQuery)
+    {
+        $set = new QueryResultset();
+        /** @var SelectQuery $query */
+        $query = $contentQuery->getService('select');
+        $query->setSingleFetchMode(false);
+
+        foreach ($contentQuery->getContentTypes() as $contentType) {
+            $contentType = str_replace('-', '_', $contentType);
+
+            $repo = $contentQuery->getContentRepository();
+            $qb = $repo->getQueryBuilder();
+            $query->setQueryBuilder($qb);
+            // $query->setContentType($contentType);
+            $query->setContentType('content');
+
+            $query->setParameters($contentQuery->getParameters());
+            $contentQuery->runScopes($query);
+            $contentQuery->runDirectives($query);
+
+            // This is required. Not entirely sure why.
```

Comment:
because it builds all the parameters in query. the name of method is wrong if it is not meaningful.
rename to e.g. buildWhereParameters()

Reply:
This pull request is based on Bolt 3's code, and the original did not use this `build` function. And I'm still not sure why it is needed for this, but not for Bolt _4_.

Reply:
Well, I don't know exactly how it worked in Bolt 3, but there was DBAL QB and here is the ORM one so it works differently ;)

Reply:
Yeah, that may be it actually 🤔